core-bundle/src/Command/UserCreateCommand.php

```diff
@@ -0,0 +1,296 @@
+        $this->framework = $framework;
+        $this->connection = $connection;
+        $this->encoderFactory = $encoderFactory;
+        $this->locales = $locales;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addOption('username', 'u', InputOption::VALUE_REQUIRED, 'The username to create')
+            ->addOption('name', null, InputOption::VALUE_REQUIRED, 'The full name')
+            ->addOption('email', null, InputOption::VALUE_REQUIRED, 'The email address')
+            ->addOption('password', 'p', InputOption::VALUE_REQUIRED, 'The password')
+            ->addOption('language', null, InputOption::VALUE_REQUIRED, 'The user language (ISO 639-1 language code)')
+            ->addOption('admin', null, InputOption::VALUE_NONE, 'Give admin permissions to the new user')
+            ->addOption('group', null, InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY, 'The groups to assign the user to')
+            ->addOption('change-password', null, InputOption::VALUE_NONE, 'Require user to change the password on the first back end login')
+            ->setDescription('Create a new Contao back end user.')
+        ;
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output): void
+    {
+        if (null === $input->getOption('username')) {
+            $username = $this->ask('Please enter the username: ', $input, $output);
+
+            $input->setOption('username', $username);
+        }
+
+        if (null === $input->getOption('name')) {
+            $name = $this->ask('Please enter the full name: ', $input, $output);
+
+            $input->setOption('name', $name);
+        }
+
+        $emailCallback = static function ($value) {
+            if (!Validator::isEmail($value)) {
+                throw new \InvalidArgumentException('The email address is invalid.');
+            }
+
+            return $value;
+        };
+
+        if (null === $input->getOption('email')) {
+            $email = $this->ask('Please enter the email address: ', $input, $output, $emailCallback);
+
+            $input->setOption('email', $email);
+        } else {
+            $emailCallback($input->getOption('email'));
+        }
+
+        if (null === $input->getOption('language')) {
+            $language = $this->askChoice('Please type the user language code: ', $this->locales, $input, $output);
+
+            $input->setOption('language', $language);
+        }
+
+        /** @var Config $config */
+        $config = $this->framework->getAdapter(Config::class);
+        $minLength = $config->get('minPasswordLength');
+        $username = $input->getOption('username');
+
+        $passwordCallback = static function ($value) use ($username, $minLength): string {
+            if ('' === trim($value)) {
+                throw new \RuntimeException('The password cannot be empty');
+            }
+
+            if (grapheme_strlen($value) < $minLength) {
```

Comment:
are you sure `grapheme_strlen` is correct here?

Reply:
Same as here https://github.com/contao/contao/blob/9ee2fbd87cb0bf4f4e8f0aca70035a902b99bd04/core-bundle/src/Resources/contao/forms/FormPassword.php#L125